deltacat/tests/compactor/test_compaction_session.py

```diff
@@ -0,0 +1,406 @@
+import ray
+from moto import mock_s3
+import pytest
+import os
+import json
+import boto3
+from typing import Any, Dict
+from typing import List
+from boto3.resources.base import ServiceResource
+import pyarrow as pa
+
+from enum import Enum
+
+
+class PartitionKeyType(str, Enum):
+    INT = "int"
+    STRING = "string"
+    TIMESTAMP = "timestamp"
+
+
+class PartitionKey(dict):
+    @staticmethod
+    def of(key_name: str, key_type: PartitionKeyType) -> PartitionKey:
+        return PartitionKey({"keyName": key_name, "keyType": key_type.value})
+
+    @property
+    def key_name(self) -> str:
+        return self["keyName"]
+
+    @property
+    def key_type(self) -> PartitionKeyType:
+        key_type = self["keyType"]
+        return None if key_type is None else PartitionKeyType(key_type)
+
+
+MAX_RECORDS_PER_FILE: int = 1
+TEST_S3_RCF_BUCKET_NAME = "test-compaction-artifacts-bucket"
+# REBASE  src = spark compacted table to create an initial version of ray compacted table
+TEST_SOURCE_NAMESPACE = "source_test_namespace"
+TEST_SOURCE_TABLE_NAME = "test_table"
+TEST_SOURCE_TABLE_VERSION = "1"
+
+TEST_DESTINATION_NAMESPACE = "destination_test_namespace"
+TEST_DESTINATION_TABLE_NAME = "destination_test_table"
+TEST_DESTINATION_TABLE_VERSION = "1"
+
+
+"""
+HELPERS
+"""
+
+
+def helper_read_s3_contents(
+    s3_resource: ServiceResource, bucket_name: str, key: str
+) -> Dict[str, Any]:
+    response = s3_resource.Object(bucket_name, key).get()
+    file_content: str = response["Body"].read().decode("utf-8")
+    return json.loads(file_content)
+
+
+"""
+MODULE scoped fixtures
+"""
+
+
+@pytest.fixture(autouse=True, scope="module")
+def mock_aws_credential():
+    os.environ["AWS_ACCESS_KEY_ID"] = "testing"
+    os.environ["AWS_SECRET_ACCESS_ID"] = "testing"
+    os.environ["AWS_SECURITY_TOKEN"] = "testing"
+    os.environ["AWS_SESSION_TOKEN"] = "testing"
+    os.environ["AWS_DEFAULT_REGION"] = "us-east-1"
+    yield
+
+
+@pytest.fixture(scope="module")
+def s3_resource(mock_aws_credential):
+    with mock_s3():
+        yield boto3.resource("s3")
+
+
+@pytest.fixture(scope="module")
+def compaction_artifacts_s3_bucket(s3_resource: ServiceResource):
+    s3_resource.create_bucket(
+        ACL="authenticated-read",
+        Bucket=TEST_S3_RCF_BUCKET_NAME,
+    )
+    yield
+
+
+"""
+Function scoped fixtures
+"""
+
+
+@pytest.fixture(scope="function")
+def ds_mock_kwargs():
+    DATABASE_FILE_PATH_KEY, DATABASE_FILE_PATH_VALUE = (
+        "db_file_path",
+        "deltacat/tests/local_deltacat_storage/db_test.sqlite",
+    )
+    # see deltacat/tests/local_deltacat_storage/README.md for documentation
+    kwargs_for_local_deltacat_storage: Dict[str, Any] = {
+        DATABASE_FILE_PATH_KEY: DATABASE_FILE_PATH_VALUE,
+    }
+    yield kwargs_for_local_deltacat_storage
+    # cleanup the created db file after
+    if os.path.exists(DATABASE_FILE_PATH_VALUE):
+        os.remove(DATABASE_FILE_PATH_VALUE)
+
+
+INCREMENTAL_TEST_CASES = {
```

Comment:
This can be its own file.

Reply:
Many comments on this file are not yet addressed.

Reply:
Done